packages/test/functional-tests/src/test/containerRuntime.spec.ts

```diff
@@ -241,5 +243,68 @@ describe("Container Runtime", () => {
             assert.strictEqual(2, batchBegin, "Did not receive correct batchBegin event for the batch");
             assert.strictEqual(2, batchEnd, "Did not receive correct batchEnd event for the batch");
         });
+
+        it("reconnects after receiving a leave op", async () => {
+            const service = new MockDocumentService(
+                undefined,
+                () => deltaConnection.disposed ? new MockDocumentDeltaConnection("test") : deltaConnection,
+            );
+
+            // overwriting connection mode to return the requested mode from client
+            // instead of default to "write" in MockDocumentDeltaConnection
+            service.connectToDeltaStream = async (newClient: IClient): Promise<IDocumentDeltaConnection> => {
+                const connection = deltaConnection.disposed ? new MockDocumentDeltaConnection("test") : deltaConnection;
```

Comment:
I'm not sure why you need to overwrite connectToDeltaStream() method, when MockDocumentService.connectToDeltaStream() only calls callback you provided on line 250 - you can simply move this logic into there.

Reply:
Is it important to return same deltaConnection? I.e. should we always (unconditionally) simply create a new instance of MockDocumentDeltaConnection?

Reply:
i chose to overwrite connectToDeltaStream because it has access to client, i needed the mode from client

Reply:
Got it. Consider adding client argument to the callback above instead - I'm sure someone else will need it in the future, so I'd rather make system more flexible and allow more streamlined usage of mock classes (i.e. have no need to overwrite methods like that)